python/ct/client/log_client.py

```diff
@@ -209,18 +213,29 @@ def __str__(self):
     def get_response(self, uri, params=None):
         """Get an HTTP response for a GET request."""
         try:
-            return requests.get(uri, params=params, timeout=self._timeout,
-                                verify=self._ca_bundle)
-        except requests.exceptions.RequestException as e:
+            return RequestHandler._build_requests_style_response(
+                self._http.request(RequestHandler._uri_with_params(uri, params)))
+        except Exception as e:
             raise HTTPError("Connection to %s failed: %s" % (uri, e))
 
     def post_response(self, uri, post_data):
         try:
-            return requests.post(uri, data=json.dumps(post_data),
-                                 timeout=self._timeout, verify=self._ca_bundle)
-        except requests.exceptions.RequestException as e:
+            return RequestHandler._build_requests_style_response(
+                self._http.request(uri, "POST", json.dumps(post_data)))
+        except Exception as e:
             raise HTTPError("POST to %s failed: %s" % (uri, e))
 
+    class Response(object):
```

Comment:
How about using namedtuple for this? https://docs.python.org/2/library/collections.html#collections.namedtuple